thrift_rpc/recorder.go

```diff
@@ -451,19 +466,25 @@ func (r *Recorder) reportLoop() {
 	}
 
 	tickerChan := time.Tick(minReportingPeriod)
-	for range tickerChan {
-		r.maybeLogInfof("reporting alarm fired")
-
-		// Kill the reportLoop() if we've been disabled.
-		r.lock.Lock()
-		if r.disabled {
+	for {
+		select {
+		case <-tickerChan:
+			r.maybeLogInfof("reporting alarm fired")
+
+			// Kill the reportLoop() if we've been disabled.
+			r.lock.Lock()
+			if r.disabled {
+				r.lock.Unlock()
+				break
+			}
 			r.lock.Unlock()
-			break
-		}
-		r.lock.Unlock()
 
-		if r.shouldFlush() {
-			r.Flush()
+			if r.shouldFlush() {
+				r.Flush()
+			}
+
+		case <-closech:
+			return
```

Comment:
See above comment regarding thrift. The thrift tracer is legacy and we do not recommend it's usage in go at this time, any new code should use the GRPC implementation. We will be adding documentation about this shortly.